File: Factorial Finder/fact.rb
require 'benchmark'

# Recursive with Tail recursion
def fhelper (x, acc)
	return acc * x if x == 2
	fhelper(x - 1, acc * x)
end

def ft_recursive (x)
	fhelper(x, 1)
end

# Iterative with each
def ft_loop (number)
    f = 1
    (1..number).each { |i| f *= i }
    f
end

# Iterative with inject
def ft_inject(n)
	(1..n).inject {|prod, i| prod * i;}
end

puts ">"*5 + "Factorial" + "<"*5
puts "Enter number: "
num = gets.chomp.to_i
bench1 = Benchmark.measure { puts ft_recursive(num) } 
bench2 = Benchmark.measure { puts ft_loop(num) }
bench3 = Benchmark.measure { puts ft_inject(num) }
puts "Recursive in #{bench1}"
puts "Iterative loop in #{bench2}"
puts "Iterative inject in #{bench3}"

File: Factorial Finder/fact.py
import time

import sys
sys.setrecursionlimit(50000)

def ft_recursive(n):
    z=1
    if n>1:
        z=n*ft_recursive(n-1)
    return z

def ft_print(func, n):
	t1 = time.time()
	print func(n)
	t2 = time.time()
	return "Completed in " + func.__name__ + " " + str(t2-t1)

print ">"*5, "Factorial", "<"*5
print "Enter number: "
n = int(raw_input("Enter a number: "))
b1 = ft_print(ft_recursive, n) #~ 21825
print b1